src/arrayMethodJoin.js

```diff
@@ -4,8 +4,21 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      result += this[i] !== null && this[i] !== undefined
+        ? this[i].toString() + separator
+        : '' + separator;
+    }
+
+    switch (true) {
```

Comment:
This switch statement with one case doesn't make sense
You can just use early return

```
if(`${separator}`.length >= 1) {
  return result.slice(0, -`${separator}`.length);
}

return result;
```